client/core/core.go

```diff
@@ -3090,6 +3098,7 @@ func (c *Core) dbTrackers(dc *dexConnection) (map[order.OrderID]*trackedTrade, e
 		if err != nil {
 			c.log.Errorf("Error setting cancel order info %s: %v", co.ID(), err)
 		}
+		delete(cancels, cancelID)
```

Comment:
It appears that you're adding to `cancels` and deleting from `cancels` in the same loop, but there's no guarantee that we'll see the cancel order before it's trade, and if the trade comes first, the cancel will be added and never deleted, and you'll end up setting valid cancels as revoked.

Reply:
Ah, good catch.  Didn't notice that.